internal/core/services/k8s.go

```diff
@@ -0,0 +1,162 @@
+	}
+
+	configData, err := ks.GetDataFromConfigMap(kuberData, deploymentData.LabelMainProcessName)
+	if err != nil {
+		ks.logger.Error("err while getting data from configMap K8sAPI %s", err)
+		return &models.DataFromK8sAPI{}, err
+	}
+
+	dataFromK8sAPI := &models.DataFromK8sAPI{
+		KuberData:      kuberData,
+		DeploymentData: deploymentData,
+		ConfigMapData:  configData,
+	}
+
+	return dataFromK8sAPI, nil
+}
+
+func (ks *KuberService) ConnectionToK8sAPI() (*models.KuberData, error) {
+	ks.logger.Info("### 🌀 Attempting to use in cluster config")
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		ks.logger.Error(err)
+		return nil, err
+	}
+
+	ks.logger.Info("### 💻 Connecting to Kubernetes API, using host: ", config.Host)
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		ks.logger.Error(err)
+		return nil, err
+	}
+
+	namespaceBytes, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
+	if err != nil {
+		ks.logger.Error(err)
+		return nil, err
+	}
+	namespace := string(namespaceBytes)
+
+	podName := os.Getenv("POD_NAME")
+
+	targetName := func(podName string) string {
+		elements := strings.Split(podName, "-")
+		newElements := elements[:len(elements)-2]
+		return strings.Join(newElements, "-")
+	}(podName)
+	if targetName == "" {
+		ks.logger.Fatalln("### 💥 Env var DEPLOYMENT_NAME was not set")
+	}
+	targetType := os.Getenv("DEPLOYMENT_TYPE")
+	kuberData := &models.KuberData{
+		Clientset:  clientset,
+		Namespace:  namespace,
+		TargetName: targetName,
+		TargetType: targetType,
+	}
+	return kuberData, nil
+}
+func (ks *KuberService) GetDataFromConfigMap(kuberData *models.KuberData, label string) (*models.ConfigMapData, error) {
+	cm, err := kuberData.Clientset.CoreV1().ConfigMaps(kuberData.Namespace).Get(context.Background(), os.Getenv("CONFIG_MAP_NAME_FOR_HASHER"), metav1.GetOptions{})
+	if err != nil {
+		ks.logger.Error("err while getting data from configMap kuberAPI ", err)
+		return nil, err
+	}
+
+	var configMapData models.ConfigMapData
+	valuesEnv := make(map[string]string)
+	for key, value := range cm.Data {
+		if key == label {
+			envs := strings.Split(strings.TrimSpace(value), "\n")
+			for _, subStr := range envs {
+				valuesEnvs := strings.Split(strings.TrimSpace(subStr), "=")
+				valuesEnv[valuesEnvs[0]] = valuesEnvs[1]
+			}
+		}
+	}
+
+	if value, ok := valuesEnv["PID_NAME"]; ok {
+		configMapData.ProcName = value
+	}
+	if value, ok := valuesEnv["MOUNT_PATH"]; ok {
+		configMapData.MountPath = value
+	}
+
+	return &configMapData, err
+}
+
+func (ks *KuberService) GetDataFromDeployment(kuberData *models.KuberData) (*models.DeploymentData, error) {
+	allDeploymentData, err := kuberData.Clientset.AppsV1().Deployments(kuberData.Namespace).Get(context.Background(), kuberData.TargetName, metav1.GetOptions{})
+	if err != nil {
+		ks.logger.Error("err while getting data from kuberAPI ", err)
+		return nil, err
+	}
+
+	deploymentData := models.DeploymentData{}
```

Comment:
Could we change it to an "inline" declaration? )
like `
deploymentData := models.DeploymentData{
NameDeployment = kuberData.TargetName,
Timestamp = fmt.Sprintf("%v", allDeploymentData.CreationTimestamp),
NamePod = os.Getenv("POD_NAME")
}
`

Reply:
Fixed it